src/modules/folder/folder.repository.ts

```diff
@@ -447,6 +449,59 @@ export class SequelizeFolderRepository implements FolderRepository {
     return folders.map((folder) => this.toDomain(folder));
   }
 
+  async calculateFolderSize(
+    folderUuid: string,
+    timeoutSeconds = 15,
+  ): Promise<number> {
+    try {
+      const calculateSizeQuery = `
+      BEGIN;
```

Comment:
`BEGIN` and `COMMIT` usage create a transaction, we want to avoid that at all costs, specially when working with frequently accessed tables like files and folders

Reply:
Yes, I know, but to be able to use a LOCAL statement_timeout, the query should be in a transaction otherwise it doesn't work.

Reply:
Here is the warning.

<img width="1287" alt="Screenshot 2024-01-16 at 11 15 19 AM" src="https://github.com/internxt/drive-server-wip/assets/145517183/0d9e8fcf-8155-4784-96b7-5be866ab74e2">

Reply:
We can not use transactions, that is going to impact the service performance badly in the short and medium term. If the timeout is not possible without transactions, let's set a limit on the number of files to fetch in total to 100,000, so the sum is significant enough.

Reply:
I can change the timeout of the global statement and when the query finishes restore it as it was before, so that it doesn't require a transaction. Please confirm which of the two ways do you implement?

Reply:
I already modified the query, what I did was create a new field as row_num in the first query and in the recusivity I was adding each iteration and put the limit with a `where statement`, something like this: `WHERE fr.row_num < 100000` , and left the timeout error handler.